src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProvider.java

```diff
@@ -41,12 +42,39 @@ public String getLatestReleaseNumber() {
         return this.allReleases.get(this.allReleases.size() - 1).toString();
     }
 
+    /**
+     * Get the latest release in the given major branch
+     *
+     * @param majorReleaseNumber Integer major release number, eg. (6) or (7)
+     * @return Latest version with that major
+     * @throws java.util.NoSuchElementException if the major does not match anything
+     */
+    public ExasolVersionNumber getLatestReleaseForMajor(int majorReleaseNumber) {
+        return this.allReleases.stream().filter(version -> version.getMajorVersion() == majorReleaseNumber)
+                .max(Comparator.naturalOrder()).orElseThrow();
```

Comment:
you could also simply pick the last once since they are sorted anyway...
But since performance does not play a role here...

Reply:
Not going to rely on the sorted nature here :)